### PR TITLE
add scatter plot props for corresponding visualization

### PR DESCRIPTION
**Pull request by moontrip**
added three new props to the scatter plot component for visualization: displayLegend, displayLibraryControls, and setMargin. These props are useful for handling the grid view of the scatter plot viz.

### CODE REVIEW

File: src/plots/ScatterAndLinePlotGeneral.tsx [hunk 1]
```diff
@@ -17,7 +17,10 @@ export interface ScatterplotProps<T extends keyof PlotData> extends PlotProps {
   //DKDK involving CI, x & y range may need to be set
   xRange?: number[] | Date[];
   yRange?: number[] | Date[];
-  showLegend?: boolean;
+  //DKDK add enable/disable legend and built-in controls
+  displayLegend?: boolean;
+  displayLibraryControls?: boolean;
+  setMargin?: { l: number; r: number; b: number; t: number };
```
Comment (dmfalke) [inline src/plots/ScatterAndLinePlotGeneral.tsx:23]: This seem redundant with the `margin` property defined [here](https://github.com/VEuPathDB/web-components/blob/84f30c59b0e9a471d21a02b373c9c336cf69b94d/src/plots/PlotlyPlot.tsx#L23). Can this be removed?

**Review by dmfalke — CHANGES_REQUESTED** [review]
See inline comment requesting a change.

File: src/plots/ScatterAndLinePlotGeneral.tsx (hunk 1 above)
Comment (moontrip) [reply src/plots/ScatterAndLinePlotGeneral.tsx]: @dmfalke Thank you for finding that redundant prop! 👍 I just remembered that actually it was me to add the margin prop at PlotlyPlot component long time ago haha. I have removed it and changed corresponding part that used setMargin prop

**Review by dmfalke — APPROVED** [review]

**Comment by dmfalke** [comment]
Thanks for making these changes, @moontrip. Feel free to merge this at your convenience.

**Pull request merged**